content/blog/ai-gateway-prompt-enrichment.md

```diff
@@ -0,0 +1,118 @@
+
+### Before you begin {#example-prereq}
+
+Make sure that you have a basic {{< reuse "docs/snippets/product-name.md" >}} environment set up with AI Gateway and your LLM provider. If you don’t, check out these docs:
+
+1. [Get started with {{< reuse "docs/snippets/product-name.md" >}}](https://kgateway.dev/docs/quickstart/)
+2. [Set up AI Gateway](https://kgateway.dev/docs/ai/setup/)
+3. [Authenticate to the LLM provider](https://kgateway.dev/docs/ai/auth/)
+
+### Step 1: Prepend a system prompt {#example-system-prompt}
+
+Create a TrafficPolicy resource to enrich your prompts and configure additional settings. The following example prepends a system prompt of `Parse the unstructured text into CSV format` to each request that is sent to the `openai` HTTPRoute.
+
+```yaml {filename="trafficpolicy.yaml"}
+kubectl apply -f- <<EOF
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: TrafficPolicy
+metadata:
+  name: openai-opt
+  namespace: kgateway-system
+  labels:
+    app: ai-kgateway
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: openai
+  ai:
+    promptEnrichment:
+      prepend:
+      - role: SYSTEM
+        content: "Parse the unstructured text into CSV format."
+EOF
+```
+
+### Step 2: Send a test request {#example-try}
+
+Now that you have prompt enrichment set up, it’s time to try out a request.
+
+Get the IP address of your AI Gateway, such as with the following command.
+
+```bash {filename="AI Gateway IP address"}
+export INGRESS_GW_ADDRESS=$(kubectl get svc -n kgateway-system ai-gateway -o jsonpath="{.status.loadBalancer.ingress[0]['hostname','ip']}")
+echo $INGRESS_GW_ADDRESS
+```
+
+As a client, send a request without a system prompt.
+
+```bash {filename="curl request"}
+curl "$INGRESS_GW_ADDRESS:8080/openai" -H content-type:application/json -d '{
+   "model": "gpt-3.5-turbo",
+   "messages": [
+     {
+       "role": "user",
+       "content": "The recipe called for eggs, flour and sugar. The price was $5, $3, and $2."
+     }
+   ]
+ }' | jq -r '.choices[].message.content'
+```
+
+Although the system prompt instructions are not in the request, the unstructured text in the user prompt is still transformed into structured CSV format. This is because the system prompt is automatically prepended by AI Gateway before it is sent to the LLM provider.
+
+```csv {filename="curl response"}
+Item, Price
+Eggs, $5
+Flour, $3
+Sugar, $2
+```
+
+## Conclusion
+
+Good work! You played around with prepending system instructions to your prompts with a simple kgateway TrafficPolicy. Now you’re ready to feed the TrafficPolicy CRD to your favorite LLM and have it help you craft the perfect prompts for your AI-powered apps! 😂 
+
+If you want to try this out for yourself, go through the accompanying [AI Gateway](kgateway.dev/resources/labs) free hands-on labs. You are always welcome toget involved in the [community](https://github.com/kgateway-dev/kgateway) and let us know how it goes.
```

Comment:
```suggestion
If you want to try this out for yourself, go through the accompanying [AI Gateway](kgateway.dev/resources/labs) free hands-on labs. You are always welcome to [get involved in the community](https://github.com/kgateway-dev/community) and let us know how it goes.
```

accompanying free AI Gateway hands-on labs?
(Opinion size age shape etc)
Not sure what is correct but it sounds wordy to me.

I also think "get involved in the community" is a different CTA than "let us know what you thought of the lab"; perhaps something more like "Come by our Slack channel and let us know what you thought" ?